homeassistant/components/tplink/entity.py

```diff
@@ -57,24 +94,193 @@ async def _async_wrap(self: _T, *args: _P.args, **kwargs: _P.kwargs) -> None:
     return _async_wrap
 
 
-class CoordinatedTPLinkEntity(CoordinatorEntity[TPLinkDataUpdateCoordinator]):
+class CoordinatedTPLinkEntity(CoordinatorEntity[TPLinkDataUpdateCoordinator], ABC):
     """Common base class for all coordinated tplink entities."""
 
     _attr_has_entity_name = True
 
     def __init__(
-        self, device: Device, coordinator: TPLinkDataUpdateCoordinator
+        self,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        feature: Feature | None = None,
+        parent: Device | None = None,
```

Comment:
I'm wondering if these should be `kwargs` only?

Reply:
Changed.